python/PresentationBuilder/images/DjangoWikiImage.py

```diff
@@ -20,50 +20,52 @@ def extractName(match):
   def __init__(self, name, params):
     ImageBase.__init__(self, name, params)
 
-    # Store the Django wiki image id
-    self._id = int(name)
-
     # Get a reference to the image map contained in DjangoSlideSet
-    self._image_map = self.parent.parent.images
+    image_settings = self.parent.parent.images[name]
 
     # Set the name and url parameters
-    name, url = self._image_map[self.name()]
     if not self.isParamValid('name'):
-      self._pars['name'] = name
+      self.parameters()['name'] = name
     if not self.isParamValid('url'):
-      self._pars['url'] = url
+      self.parameters()['url'] = image_settings['url']
+
+    # Apply the settings from the DjangoSlideSet
+    for pair in image_settings['settings'].split():
+      k,v = pair.strip().split(':')
+      if k in params:
+        params[k] = v
 
   ##
   # Performs the regex matching for Django images
   @staticmethod
   def match(markdown):
 
-    # List of match iterators to return
-    m = dict()
+    # This list to be output
+    output = []
+
+    # A list of ids to avoid outputting the same image twice
+    ids = []
 
     # Caption
     pattern = re.compile(r'\s*\[image:([0-9]*)(.*)\]\s*\n\s{4,}(.*?)\n')
```

Comment:
When your RegExs get this big and nasty there are two things you can do to clean them up.
 - Use the re.X flag to space things out and to add comments right inside the expression
 - Use named captures